ReservedNamesCheck exception #define _ ...

Hi there,

The new plugin is throwing exception when I make a alias to a field in a struct named as "_".
[code]
#undef _
#define _ ks._
[/code]

The exception threw:
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: 1
	at java.lang.String.charAt(String.java:658)
	at org.sonar.cxx.checks.ReservedNamesCheck.visitFile(ReservedNamesCheck.java:73)
	at com.sonar.sslr.impl.ast.AstWalker.walkAndVisit(AstWalker.java:67)
	at org.sonar.squidbridge.AstScanner.scanFiles(AstScanner.java:106)
	... 35 more

--- a/cxx-checks/src/main/java/org/sonar/cxx/checks/ReservedNamesCheck.java
+++ b/cxx-checks/src/main/java/org/sonar/cxx/checks/ReservedNamesCheck.java
@@ -70,7 +70,7 @@ public class ReservedNamesCheck extends SquidCheck<Grammar> implements CxxCharse
       String[] sub = line.split("^\\s*#define\\s+", 2);
       if (sub.length>1) {
         String name = sub[1].split("[\\s(]",2)[0];
-        if (name.startsWith("_") && Character.isUpperCase(name.charAt(1))) {
+        if (name.startsWith("_") && name.length() > 1 && Character.isUpperCase(name.charAt(1))) {
           getContext().createLineViolation(this, "Reserved name used for macro (begins with underscore followed by a capital letter)", nr);
         }
         else if (name.contains("__")) {

--- a/cxx-checks/src/test/resources/checks/ReservedNamesCheck.cc
+++ b/cxx-checks/src/test/resources/checks/ReservedNamesCheck.cc
@@ -23,6 +23,7 @@ public:
 #define _test
 #define TRUE_TEST
 /*#define TRUE this is not an error...*/
+#define _ ks._
 
 int main()
 {

